src/command/dev/watcher.rs

```diff
@@ -191,30 +204,38 @@ impl SubgraphSchemaWatcher {
                 match last_message {
                     Some(last_message) => {
                         if &subgraph_definition.sdl != last_message {
+                            if self.subgraph_retry_countdown < self.subgraph_retries {
+                                eprintln!(
+                                    "{} subgraph connectivity restored for {}",
+                                    Emoji::Reload,
+                                    self.subgraph_key.0
+                                )
+                            }
                             self.message_sender.update_subgraph(&subgraph_definition)?;
                         }
                     }
                     None => {
                         self.message_sender.add_subgraph(&subgraph_definition)?;
                     }
                 }
+                self.subgraph_retry_countdown = self.subgraph_retries;
```

Comment:
__Nit__: Should we document the logic on how `retry_countdown` works?

For my understanding:

1. It is normally at `0`
2. On success, we re-initialise it to `subgraph_retries`
3. On every attempt, we decrement until it reaches `0`

This means that if the first attempt is a failure, we stay at `0`, thus go into the branch at line 225